components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -165,6 +169,12 @@ public boolean doPreSetUserClaimValuesWithID(String userID, Map<String, String>
         Map<String, String> scimToLocalMappings = SCIMCommonUtils.getSCIMtoLocalMappings();
         String modifiedLocalClaimUri = scimToLocalMappings.get(SCIMConstants.CommonSchemaConstants.LAST_MODIFIED_URI);
         claims.put(modifiedLocalClaimUri, lastModifiedDate);
+
+        // Change dateOfBirth value to datetime format if dob is in yyyy-mm-dd format.
+        if (claims.containsKey(DATE_OF_BIRTH_LOCAL_CLAIM) &&
```

Comment:
what happens if the following condition is not matched. then, won't it store again?  We have to throw an error in that case. 

claims.get(DATE_OF_BIRTH_LOCAL_CLAIM).matches(DATE_OF_BIRTH_REGEX)

Reply:
logic changed in  7b3fb87